engine/src/main/java/org/hibernate/search/v6poc/engine/impl/SearchMappingRepositoryBuilderImpl.java

```diff
@@ -279,6 +269,40 @@ void createAndAddMapper(Map<MappingKey<?>, Mapper<?>> mappers,
 			}
 		}
 
+		void createAndAddMapping(Map<MappingKey<?>, MappingImplementor<?>> mappings) {
+			if ( mappings.containsKey( mappingKey ) ) {
+				throw new AssertionFailure(
+						"Found two mapping initiators using the same key."
+						+ " There is a bug in the mapper, please report it."
+				);
+			}
+
+			try {
+				MappingImplementor<M> mapping = mapper.build();
+				mappings.put( mappingKey, mapping );
+			}
+			catch (MappingAbortedException e) {
+				ContextualFailureCollector failureCollector = buildContext.getFailureCollector();
+				Throwable cause = e.getCause();
+				if ( cause != null ) {
+					failureCollector.add( cause );
+				}
+				Throwable[] suppressed = e.getSuppressed();
+				for ( Throwable throwable : suppressed ) {
+					failureCollector.add( throwable );
+				}
```

Comment:
I don't understand the purpose of the code above going through the cause and suppressed exceptions. It seems weird to add several failures for one original failure and AFAICS you don't provide any cause to `MappingAbortedException`.

Reply:
`MappingAbortedException` is SPI, so theoretically someone else could be throwing it, but mainly it's to be safe in case we add a cause/suppressed exception to `MappingAbortedException` in the future. Missing exceptions in a stack trace are the worst.
I added a comment to clarify that.